libyul/optimiser/LoadResolver.cpp

```diff
@@ -76,3 +105,61 @@ void LoadResolver::tryResolve(
 			if (inScope(*value))
 				_e = Identifier{locationOf(_e), *value};
 }
+
+void LoadResolver::tryEvaluateKeccak(
+	Expression& _e,
+	std::vector<Expression> const& _arguments
+)
+{
+	GasMeter gasMeter{
+		dynamic_cast<EVMDialect const&>(m_dialect),
+		!m_expectedExecutionsPerDeployment,
+		m_expectedExecutionsPerDeployment ? *m_expectedExecutionsPerDeployment : 1
+	};
+
+	auto costOfKeccak = gasMeter.costs(_e);
```

Comment:
I'm sorry but I really want to know if this is a `u256` or a `uint64_t`.

Reply:
Ok, I looked it up and it is a `size_t`. I'm wondering: Is that value large enough? Shouldn't we change `GasMeter::costs` to return a u256 instead?

Reply:
Making it u256 makes sense. However, EIP-1985 proposes an upper bound of `2**63 - 1` for `gas()`, still it `u256` would be a safe option for us.

Anyway, this change should be a separate PR.

Reply:
It's unrelated to `gas()`: The issue here is that this value is the product of the gas costs and the runs parameter.

Reply:
I acutally think it should be a `bigint` just as the numbers returned by the other gas meter.